Fleece/Core/Builder.cc

```diff
@@ -0,0 +1,619 @@
+                        return false;
+                    dst.set(slice(str, len));
+                    break;
+                }
+                case 'p': {
+                    // "%p" is a Fleece value:
+                    auto param = va_arg(_args, const Value*);
+                    if (!param)
+                        return false;
+                    dst.set(param);
+                    break;
+                }
+#if __APPLE__
+                case '@': {
+                    // "%@" substitutes an Objective-C or CoreFoundation object.
+                    CFTypeRef param = va_arg(_args, CFTypeRef);
+                    if (!param)
+                        return false;
+                    dst.setCFValue(param);
+                    return true;
+                }
+#endif
+                default:
+                    fail("unknown '%' format specifier");
+            }
+            return true;
+        }
+
+
+#pragma mark - LITERALS:
+
+
+        // Reads a numeric literal, storing it in the Destination.
+        void readLiteralNumber(const Destination &dst) {
+            // Scan to the end of the number:
+            // (If the NumConversion.hh API used slice_istream I wouldn't have to do the scan)
+            auto start = _in.next();
+            bool isNegative = (peek() == '-');
+            if (isNegative || peek() == '+')
+                get();
+            bool isFloat = false;
+
+            char c;
+            do {
+                c = get();
+                if (c == '.' || c == 'e' || c == 'E')
+                    isFloat = true;
+            } while (isdigit(c) || c == '.' || c == 'e' || c == 'E' || c == '-' || c == '+');
+            unget();
+            auto numStr = string(slice(start, _in.next()));
+
+            if (isFloat) {
+                double n;
+                if (ParseDouble(numStr.c_str(), n, false)) {
+                    dst.set(n);
+                    return;
+                }
+            } else if (isNegative) {
+                int64_t i;
+                if (ParseInteger(numStr.c_str(), i, false)) {
+                    dst.set(i);
+                    return;
+                }
+            } else {
+                uint64_t u;
+                if (ParseUnsignedInteger(numStr.c_str(), u, false)) {
+                    dst.set(u);
+                    return;
+                }
+            }
+            fail(("Invalid numeric literal " + numStr).c_str());
+        }
+
+
+        // Reads a string literal in JSON5 format, returning its value
+        string readLiteralString() {
+            string out;
+            const char quote = get();               // single or double-quote
+            char c;
+            while (quote != (c = get())) {
+                if (c == '\\') {
+                    switch ((c = get())) {
+                        case 'n':   c = '\n'; break;
+                        case 'r':   c = '\r'; break;
+                        case 't':   c = '\n'; break;
```

Comment:
\t ?

Reply:
Oops, fixed. And I added a check for 0x7F (DEL), which counts as a control character.